packages/salesforcedx-apex-debugger/src/adapter/apexDebug.ts

```diff
@@ -438,6 +522,13 @@ export class ApexDebug extends DebugSession {
         this.logEvent(message);
         this.requestThreads.splice(threadId, 1);
         this.sendEvent(new ThreadEvent('exited', threadId));
```

Comment:
When user clicks 'continue', the debug adapter sends a run command to server. That request will either finish or hit another breakpoint. If it finishes, we're supposed to send a ThreadEvent('exited') to VS Code so it knows to stop tracking that thread. For some reason, VS Code loses track of all stopped threads (apex debugger can have up to 2). The current workaround is to send stopped events for threads that are still stopped so VS Code can continue to track it. The downside is it makes VS Code ask the debug adapter for stack trace of those threads, which is wasteful because they haven't changed.

Reply:
Could you include a reference to the issue so that I can keep track?
You can just paste it as plain text so that GitHub won't auto-link the two.

Reply:
`https://github.com/Microsoft/vscode-debugadapter-node/issues/130`

Reply:
@DatGuyJonathan Did you see the last reply? They closed https://github.com/Microsoft/vscode/issues/33839  as "as-designed". Is the additional stop event still necessary with `allThreadsContinued: false`?

Reply:
Yes, I already have that in my handling of continuedRequest. I'll keep the additional stop event for now and continue the conversation with them.